bin/reth/src/commands/bitfinity_reset_evm_state.rs

```diff
@@ -0,0 +1,202 @@
+use reth_node_core::args::BitfinityResetEvmStateArgs;
+use reth_node_core::dirs::{DataDirPath, MaybePlatformPath};
+use reth_primitives::{StorageEntry, B256};
+use reth_provider::{BlockNumReader, BlockReader, ProviderFactory};
+use tracing::{debug, info};
+
+/// Builder for the `bitfinity reset evm state` command
+#[derive(Debug, Parser)]
+pub struct BitfinityResetEvmStateCommandBuilder {
+    /// The path to the data dir for all reth files and subdirectories.
+    ///
+    /// Defaults to the OS-specific data directory:
+    ///
+    /// - Linux: `$XDG_DATA_HOME/reth/` or `$HOME/.local/share/reth/`
+    /// - Windows: `{FOLDERID_RoamingAppData}/reth/`
+    /// - macOS: `$HOME/Library/Application Support/reth/`
+    #[arg(long, value_name = "DATA_DIR", verbatim_doc_comment, default_value_t)]
+    pub datadir: MaybePlatformPath<DataDirPath>,
+
+    /// Bitfinity Related Args
+    #[clap(flatten)]
+    pub bitfinity: BitfinityResetEvmStateArgs,
+}
+
+impl BitfinityResetEvmStateCommandBuilder {
+
+    /// Build the command
+    pub async fn build(self) -> eyre::Result<BitfinityResetEvmStateCommand> {
+        let evm_datasource_url = self.bitfinity.evm_datasource_url.clone();
+        info!(target: "reth::cli", "Fetching chain spec from: {}", evm_datasource_url);
+        let chain = Arc::new(BitfinityEvmClient::fetch_chain_spec(evm_datasource_url).await?);
+
+        let data_dir = self.datadir.unwrap_or_chain_default(chain.chain);
+        let db_path = data_dir.db();
+        let db = Arc::new(init_db(db_path, Default::default())?);
+        let provider_factory = ProviderFactory::new(db.clone(), chain, data_dir.static_files())?;
+
+        Ok(BitfinityResetEvmStateCommand::new(provider_factory, self.bitfinity))
+    }
+}
+
+/// Command that initializes the reset of remote EVM node using the current node state
+#[derive(Debug)]
+pub struct BitfinityResetEvmStateCommand {
+    provider_factory: ProviderFactory<Arc<DatabaseEnv>>,
+    bitfinity: BitfinityResetEvmStateArgs,
+}
+
+impl BitfinityResetEvmStateCommand {
+
+    /// Create a new instance of the command
+    pub fn new(provider_factory: ProviderFactory<Arc<DatabaseEnv>>, bitfinity: BitfinityResetEvmStateArgs) -> Self {
+        Self { provider_factory, bitfinity }
+    }
+
+    /// Execute the command
+    pub async fn execute(&self) -> eyre::Result<()> {
+        let principal = candid::Principal::from_text(self.bitfinity.evmc_principal.as_str())?;
+
+        let evm_client = EvmCanisterClient::new(
+            IcAgentClient::with_identity(
+                principal,
+                self.bitfinity.ic_identity_file_path.clone(),
+                &self.bitfinity.evm_network,
+                None,
+            )
+            .await?,
+        );
+
+        let provider = self.provider_factory.provider()?;
+        let last_block_number = provider.last_block_number()?;
+        let last_block =
+            provider.block_by_number(last_block_number)?.expect("Block should be present");
```

Comment:
Just a quick question; Why didn't we disable evm first ? and also why don't we add a check that make sure is fully synced with the Bitfinity EVM ?  otherwise we might have a problem.